### DISCUSSION TITLE
How to get the balance of an address at a given block?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#105] [Questions] [answered]
Originally asked [here](https://github.com/aptos-labs/aptos-core/issues/12110). I wonder do we have an indexer API that can query history balance. Maybe there is? Petra is building a wallet dashboard and it displays the account balance over time?

**Comment by 0xbe1** [comment, marked as answer]
call view function [`0x1::coin::balance`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move#L254) using [aptos ts sdk](https://github.com/aptos-labs/aptos-ts-sdk) passing in the [`ledgerVersion` parameter](https://github.com/aptos-labs/aptos-ts-sdk/blob/53761923cb7969ef26aba6ec3e893bf38069871f/src/api/general.ts#L141)
